internal/controller/numaflowcontroller/numaflowcontroller_controller.go

```diff
@@ -17,53 +17,98 @@ limitations under the License.
 package numaflowcontroller
 
 import (
+	"bytes"
 	"context"
 	"fmt"
+	"html/template"
+	"io"
+	"strings"
+	"time"
 
+	"github.com/argoproj/gitops-engine/pkg/diff"
+	gitopsSync "github.com/argoproj/gitops-engine/pkg/sync"
+	gitopsSyncCommon "github.com/argoproj/gitops-engine/pkg/sync/common"
+	kubeUtil "github.com/argoproj/gitops-engine/pkg/utils/kube"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/schema"
+	yamlserializer "k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/record"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	runtimecontroller "sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/source"
+	sigsyaml "sigs.k8s.io/yaml"
 
+	"github.com/numaproj/numaplane/internal/common"
+	"github.com/numaproj/numaplane/internal/controller/config"
+	"github.com/numaproj/numaplane/internal/sync"
+	"github.com/numaproj/numaplane/internal/util/kubernetes"
 	"github.com/numaproj/numaplane/internal/util/logger"
 	"github.com/numaproj/numaplane/internal/util/metrics"
 	apiv1 "github.com/numaproj/numaplane/pkg/apis/numaplane/v1alpha1"
 )
 
 const (
-	ControllerNumaflowController = "numaflowcontroller-controller"
+	ControllerNumaflowController       = "numaflowcontroller-controller"
+	NumaflowControllerDeploymentName   = "numaflow-controller"
+	DefaultNumaflowControllerImageName = "numaflow"
 )
 
 // NumaflowControllerReconciler reconciles a NumaflowController object
 type NumaflowControllerReconciler struct {
 	client        client.Client
 	scheme        *runtime.Scheme
+	restConfig    *rest.Config
+	rawConfig     *rest.Config
+	kubectl       kubeUtil.Kubectl
+	stateCache    sync.LiveStateCache
 	customMetrics *metrics.CustomMetrics
 	// the recorder is used to record events
 	recorder record.EventRecorder
 }
 
 func NewNumaflowControllerReconciler(
-	c client.Client,
-	s *runtime.Scheme,
+	client client.Client,
+	scheme *runtime.Scheme,
+	rawConfig *rest.Config,
+	kubectl kubeUtil.Kubectl,
 	customMetrics *metrics.CustomMetrics,
 	recorder record.EventRecorder,
-) *NumaflowControllerReconciler {
+) (*NumaflowControllerReconciler, error) {
+	stateCache := sync.NewLiveStateCache(rawConfig, customMetrics)
+	numaLogger := logger.GetBaseLogger().WithName("state cache").WithValues("numaflowcontroller")
```

Comment:
Curious about the logger name?

Reply:
I kept it from this: https://github.com/numaproj/numaplane/blob/9d05eb5a74449dfb08e0077da23bf3230e807937/internal/controller/numaflowcontrollerrollout/numaflowcontrollerrollout_controller.go#L98
I could remove it though, I'm also unsure about it's purpose.

Reply:
Found that it is used by the state cache for logging. Moving the name in the state cache directly.